### PR TITLE
Fix federation pr job on k/f repo

### PR DESCRIPTION
**Pull request by shashidharatd**
This PR contains some fixes in-order that federation pre-submit job runs successfully.
The PR job stopped working after all federation related cluster scripts are removed from k/k repo. But these federation specific changes in cluster scripts are preserved in k/f repo. So we are making the distinction in test-infra's kubetest to use the scripts from k/f repo for federation jobs.

This PR also removes the daily recycle job and integrates the k8s cluster bring-up in the pre-submit job itself. Although the time for job completion is little longer now, it will reduce the complexity of maintaining additional job for cluster recycling. Also since federation code is separated now and the number of PR's would be comparatively lesser than on k/k repo.

/assign @krzyzacy 
/cc @irfanurrehman @marun @madhusudancs @kubernetes/sig-multicluster-pr-reviews

### CODE REVIEW

**Comment by krzyzacy** [comment]
so that's the same script copied from kubernetes repo?

**Comment by shashidharatd** [comment]
> so that's the same script copied from kubernetes repo?
yes @krzyzacy, but before the federation related code is removed from those scripts.

This will be temporary until we move k8s deployment to kubernetes-anywhere.

**Comment by krzyzacy** [comment]
@shashidharatd ok :-) can you add a todo then?

**Comment by shashidharatd** [comment]
@krzyzacy, Added an todo. PTAL

File: kubetest/bash.go [hunk 1]
```diff
@@ -48,11 +48,26 @@ func (b *bashDeployer) Up() error {
 		}
 		defer pop()
 	}
-	return finishRunning(exec.Command("./hack/e2e-internal/e2e-up.sh"))
+
+	// TODO: Remove below logic of choosing the scripts to run from federation repo once the
```
Comment (krzyzacy) [inline kubetest/bash.go]: nit: TODO(someone):

**Comment by krzyzacy** [comment]
/lgtm
/hold
feel free to `/hold cancel` 
I'll roll the kubetest change into https://github.com/kubernetes/test-infra/pull/5375

File: kubetest/e2e.go [hunk 2]
```diff
@@ -478,7 +478,12 @@ func defaultDumpClusterLogs(localArtifactsDir, logexporterGCSPath string) error
 }
 
 func dumpFederationLogs(location string) error {
-	logDumpPath := "./federation/cluster/log-dump.sh"
+	var logDumpPath string
```
Comment (krzyzacy) [inline kubetest/e2e.go]: same todo here as well?

**Comment by shashidharatd** [comment]
@krzyzacy, have added the todos as suggested.
/hold cancel
Thanks !

**Comment by shashidharatd** [comment]
@krzyzacy, this pr is merged now. can you please include this changes in latest kubekins image. Thanks !

**Comment by krzyzacy** [comment]
yep, doing that

**Pull request merged**